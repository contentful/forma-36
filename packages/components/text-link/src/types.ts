export type TextLinkVariant =
  | 'negative'
  | 'positive'
  | 'primary'
  | 'secondary'
  | 'muted'
  | 'white';
