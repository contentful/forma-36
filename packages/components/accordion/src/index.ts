export { Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';
export { AccordionHeader } from './AccordionHeader/AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader/AccordionHeader';
export { AccordionItem } from './AccordionItem/AccordionItem';
export type { AccordionItemProps } from './AccordionItem/AccordionItem';
export { AccordionPanel } from './AccordionPanel/AccordionPanel';
export type { AccordionPanelProps } from './AccordionPanel/AccordionPanel';
