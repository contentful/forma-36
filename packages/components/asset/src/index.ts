export { Asset } from './Asset';
export type { AssetProps } from './Asset';
