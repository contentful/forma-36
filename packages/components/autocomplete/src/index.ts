export { Autocomplete } from './Autocomplete';
export type { AutocompleteProps } from './Autocomplete';
