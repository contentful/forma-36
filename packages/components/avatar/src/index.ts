export { Avatar } from './Avatar';
export type { AvatarProps } from './Avatar';
export { AvatarGroup } from './AvatarGroup';
export type { AvatarGroupProps } from './AvatarGroup';
