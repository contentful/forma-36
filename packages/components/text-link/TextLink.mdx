---
title: 'TextLink'
type: 'component'
status: 'stable'
slug: /components/text-link/
github: 'https://github.com/contentful/forma-36/tree/forma-v4/packages/components/text-link'
storybook: 'https://v4-f36-storybook.netlify.app/?path=/story/components-textlink--basic'
typescript: ./src/TextLink.tsx
---

TextLinks communicate actions and linked resources to users.

## How to use TextLinks

- TextLinks are used to display actions that have a lower priority compared to actions that are accessible via a button.
- They should also be used when linking to URLs, and are preferred over using buttons when used for this purpose.
- Use clear, succinct, and accurate labels
- Start TextLink labels with verbs, for example "Add field", not "New field"
- Reduce complexity by using a small number of actions. Too many actions can create confusion when having to decide

## Component variations

There are a number of variations of TextLink styles, here is a guide for when to use them:

- **Primary** - Used for primary actions, or the default for links to URLs.
- **Positive** - Used for a positive actions, such as creating or publishing a new entity.
- **Negative** - Used for destructive actions - when something can't be undone. For example, deleting entities.
- **Secondary** - For actions that should be emphasized less than the default primary style.
- **Muted** - For actions that should be emphasized less than the secondary style.
- **White** - For actions appearing on a dark background.

```jsx
<Stack flexDirection="column">
  <TextLink variant="primary">Primary</TextLink>
  <TextLink variant="positive">Positive</TextLink>
  <TextLink variant="negative">Negative</TextLink>
  <TextLink variant="muted">Muted</TextLink>
  <TextLink variant="secondary">Secondary</TextLink>
</Stack>
```

## Code examples

Text links communicate actions and highlight linked resources to users.

```jsx
<TextLink variant="primary">Text link</TextLink>
```

### TextLink with icon (`icon`)

```jsx
<Stack flexDirection="column">
  <TextLink icon={<ExternalLinkIcon />}>Primary</TextLink>
  <TextLink icon={<ExternalLinkIcon />} alignIcon="end">
    Primary
  </TextLink>
</Stack>
```

## Content recommendations

- To make `TextLink` action-oriented, use a verb. For example, "Add field", not "New field"
- If a `TextLink` is used to navigate to a new page, clearly communicate what the user will see when they click on the link. For example, “View teams documentation” vs “Learn about teams”. Avoid generic descriptions like "click here" or "go to".
- Don't use the page's URL as the link description unless you have a good reason to do so, such as referencing a site's new address.
- Use clear and succinct copy. Aim for a few words or a short phrase.
- If a `TextLink` is used in a regular text and it's at the end of a sentence, don’t add a period inside `TextLink` content. Place it after component.
- If a `TextLink` is used as an action-oriented element, don’t add period at all, neither in component content nor after.

## Accessibility

- Avoid using extra styling that that make link looks just like regular text.
- If you use `TextLink` for linking to third party content and use `target="_blank"` prop, it is recommended to set `rel="noreferrer noopener"`. [More details about external links security](https://web.dev/external-anchors-use-rel-noopener/)

## Props of TextLink

import { Props } from '@contentful/f36-docs-utils';

<Props of="TextLink" />
