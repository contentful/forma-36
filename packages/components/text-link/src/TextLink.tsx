import React from 'react';
import { cx } from 'emotion';
import { Flex } from '@contentful/f36-core';
import type {
  CommonProps,
  PolymorphicProps,
  PolymorphicComponent,
  ExpandProps,
} from '@contentful/f36-core';
import { styles } from './TextLink.styles';
import { TextLinkVariant } from './types';

const TEXT_LINK_DEFAULT_TAG = 'a';

interface TextLinkInternalProps extends CommonProps {
  children?: React.ReactNode;
  /**
   * Determines style variation of TextLink component
   * @default primary
   */
  variant?: TextLinkVariant;
  /**
   * Disabled interaction and applies disabled styles
   * @default false
   */
  isDisabled?: boolean;
  /**
   * Expects any of the icon components
   */
  icon?: React.ReactElement;
  /**
   * Determines the icon position regarding the link text
   * @default start
   */
  alignIcon?: 'start' | 'end';
  /**
   * The element used for the root node.
   * @default a
   */
  as?: 'a' | 'button';
}

export type TextLinkProps<
  E extends React.ElementType = typeof TEXT_LINK_DEFAULT_TAG
> = PolymorphicProps<TextLinkInternalProps, E, 'disabled'>;

function _TextLink<E extends React.ElementType = typeof TEXT_LINK_DEFAULT_TAG>(
  props: TextLinkProps<E>,
  ref: React.Ref<any>,
) {
  const {
    children,
    className,
    testId = 'cf-ui-text-link',
    variant = 'primary',
    href,
    icon,
    alignIcon = 'start',
    isDisabled,
    as = TEXT_LINK_DEFAULT_TAG,
    ...otherProps
  } = props;

  const commonProps = {
    ref,
    className: cx(
      styles.textLink({
        variant,
        isDisabled,
      }),
      props.className,
    ),
    ['data-test-id']: testId,
    ...otherProps,
  };

  const iconContent = icon ? (
    <Flex as="span">
      {React.cloneElement(icon, {
        className: cx(icon.props.className, styles.textLinkIcon()),
        size: 'small',
      })}
    </Flex>
  ) : null;

  const commonContent = (
    <>
      {icon && alignIcon === 'start' && iconContent}
      {children && (
        <span
          className={styles.textLinkText({
            alignIcon: icon ? alignIcon : undefined,
          })}
        >
          {children}
        </span>
      )}
      {icon && alignIcon === 'end' && iconContent}
    </>
  );

  if (as === 'button') {
    return (
      <button {...commonProps} disabled={isDisabled} type="button">
        {commonContent}
      </button>
    );
  }

  return (
    <a
      {...commonProps}
      onClick={
        isDisabled
          ? (e) => {
              e.preventDefault();
            }
          : commonProps.onClick
      }
      href={href}
      {...(isDisabled ? { tabIndex: -1 } : {})}
    >
      {commonContent}
    </a>
  );
}

_TextLink.displayName = 'TextLink';

export const TextLink: PolymorphicComponent<
  ExpandProps<TextLinkInternalProps>,
  typeof TEXT_LINK_DEFAULT_TAG,
  'disabled'
> = React.forwardRef(_TextLink);
