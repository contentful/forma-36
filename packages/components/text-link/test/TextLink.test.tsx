import React from 'react';
import { render } from '@testing-library/react';
import { TextLink } from '../src/TextLink';

describe('TextLink', function () {
  it('renders', () => {
    const tree = render(<TextLink>hello world</TextLink>);

    expect(tree).toBeTruthy();
  });
});
