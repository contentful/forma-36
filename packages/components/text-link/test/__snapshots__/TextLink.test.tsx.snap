// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink allows passing a test id 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
  data-test-id="test-id"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink allows passing additional class names to component 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0 testing"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink allows passing additional props not consumed by component 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
  data-test-id="Testing Id"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink calls an onClick function 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<button
  class="emotion-0"
  type="button"
>
  <span>
    Text Link
  </span>
</button>
`;

exports[`TextLink prevents onClick function from being called when disabled 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: not-allowed;
  opacity: 0.5;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<button
  class="emotion-0"
  disabled=""
  type="button"
>
  <span>
    Text Link
  </span>
</button>
`;

exports[`TextLink renders as a "muted" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #AEC1CC;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #67728A;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a "negative" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #BD002A;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #990017;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a "positive" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #006D23;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #00550C;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a "primary" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a "secondary" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #5A657C;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #414D63;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a "white" link type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #ffffff;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #F7F9FA;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders as a button 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<button
  class="emotion-0"
  type="button"
>
  <span>
    Text Link
  </span>
</button>
`;

exports[`TextLink renders as a link 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-0:hover,
.emotion-0:focus {
  color: #0041AB;
}

.emotion-0:focus,
.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

<a
  class="emotion-0"
  href="#"
>
  <span>
    Text Link
  </span>
</a>
`;

exports[`TextLink renders with an icon 1`] = `
.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-3:hover,
.emotion-3:focus {
  color: #0041AB;
}

.emotion-3:focus,
.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

.emotion-2 {
  margin-left: 0.125rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  display: inline-block;
  fill: #0059C8;
  height: 16px;
  width: 16px;
  fill: currentColor;
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
}

<a
  class="emotion-3"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      ,
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    class="emotion-2"
  >
    Text Link
  </span>
</a>
`;

exports[`TextLink renders with an icon aligned right to the text 1`] = `
.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  white-space: normal;
  text-align: left;
  cursor: pointer;
  opacity: 1;
  color: #0059C8;
}

.emotion-3:hover,
.emotion-3:focus {
  color: #0041AB;
}

.emotion-3:focus,
.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3:focus {
  box-shadow: 0px 0px 0px 3px #98CBFF;
  outline: none;
  border-radius: 4px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  display: inline-block;
  fill: #0059C8;
  height: 16px;
  width: 16px;
  fill: currentColor;
  -webkit-transition: fill 0.1s ease-in-out;
  transition: fill 0.1s ease-in-out;
}

.emotion-0 {
  margin-right: 0.125rem;
}

<a
  class="emotion-3"
>
  <span
    class="emotion-0"
  >
    Text Link
  </span>
  <span
    class="emotion-2"
  >
    <svg
      class="emotion-1"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      ,
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</a>
`;
